param/__init__.py

```diff
@@ -1359,11 +1359,11 @@ class List(Parameter):
     items in the list are checked to be of that type.
     """
 
-    __slots__ = ['class_','bounds']
+    __slots__ = ['bounds', 'item_type']
 
-    def __init__(self,default=[],class_=None,instantiate=True,
+    def __init__(self,default=[],class_=None,item_type=None,instantiate=True,
                  bounds=(0,None),**params):
-        self.class_ = class_
```

Comment:
@jbednar @philippjfr Think it is worth adding a warning to tell people to use `item_type` here instead if they set `class_` or would that be too annoying/noisy?

Reply:
Too annoying.